talk/morelanguage/exceptions.tex

```diff
@@ -4,185 +4,166 @@
   \frametitlecpp[98]{Exceptions}
   \begin{block}{The concept}
     \begin{itemize}
-    \item exceptional event breaking linearity of the code
-    \item will be handled in dedicated place
+      \item to handle \textit{exceptional} events that cannot be foreseen
+      \item to cleanly jump to the place where the error can be handled
```

Comment:
```suggestion
      \item to cleanly jump to a place where the error can be handled
```
I like to propose this small distinction, because in many cases, we don't know where an error will be handled.

Reply:
I'm not convinced - although it's a minor point the programmer should know where the exception is going to be caught as part of the code design (and the compiler definitely knows 😄)

Reply:
Not really. When I throw in my library, I have no clue which other library/application includes me as a third-party dependency and whether they can handle that. I can just document and hope for the best.
Do you know what exceptions or whether `boost::filesystem::exists()` can throw?  Do the authors of Boost.Filesystem now whether your code will catch or not, or what types it tries to catch? Sadly, not everybody inherits from `std::exception` too. We live in a world where software is build from many dependencies that we don't control. They could throw anything (even int or std::string).

And certainly the compiler does not know in many cases, e.g. if you load a shared object that you pulled form the internet or that was part of a commercial/closed-source software.

Reply:
There is documentation, but anyway, this is a minor issue so I changed it.